src/wp-includes/functions.wp-scripts.php

```diff
@@ -168,9 +168,12 @@ function wp_add_inline_script( $handle, $data, $position = 'after' ) {
  *                                    If set to null, no version is added.
  * @param bool             $in_footer Optional. Whether to enqueue the script before `</body>` instead of in the `<head>`.
  *                                    Default 'false'.
+ * @param array            $attr      Optional.  Accepts an array of key value pairs.
+ *                                    Supports a `strategy` key with potential values of 'blocking', 'async' or 'defer'.
+ * @since 6.2.0 Added the `$attr` parameter.
  * @return bool Whether the script has been registered. True on success, false on failure.
  */
-function wp_register_script( $handle, $src, $deps = array(), $ver = false, $in_footer = false ) {
+function wp_register_script( $handle, $src, $deps = array(), $ver = false, $in_footer = false, $attr = false ) {
```

Comment:
this is currently implemented as a separate parameter which needs to be adjusted to match the design doc which instead overrides the `in_footer` attribute